contrib/olivere/elastic/elastictrace.go

```diff
@@ -80,7 +81,9 @@ func (t *httpTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 		if err != nil {
 			snip = http.StatusText(res.StatusCode)
 		}
-		span.SetTag(ext.Error, errors.New(snip))
+		if globalconfig.IsHTTPClientError(res.StatusCode) {
+			span.SetTag(ext.Error, errors.New(snip))
+		}
```

Comment:
This check should probably be on line 78 (`} else if res.StatusCode < 200 || res.StatusCode > 299 {`) rather than here. Like this we're checking the status code twice using two different methods.

Does it even make sense to use `IsHTTPClientError` here? This isn't a client for some random HTTP server that a customer may want to define error codes for. It's connecting to ElasticSearch, which should have well-defined errors.